### PR TITLE
Add type information to autofill service

### PR DESCRIPTION
**Pull request by MarkKoester**
Autofill service has a lot of areas that are completely missing type information. This PR adds types to most variables in the AutofillService class.

### CODE REVIEW

File: src/popup/vault/current-tab.component.ts [hunk 1]
```diff
@@ -36,12 +36,26 @@ import { Utils } from 'jslib/misc/utils';
 
 const BroadcasterSubscriptionId = 'CurrentTabComponent';
 
+export interface IPageDetail {
```
Comment (kspearrin) [inline src/popup/vault/current-tab.component.ts]: This should be in the autofill service abstraction

File: src/services/abstractions/autofill.service.ts [hunk 2]
```diff
@@ -1,7 +1,20 @@
+import { CipherView } from 'jslib/models/view';
+import { IPageDetail } from 'src/popup/vault/current-tab.component';
```
Comment (kspearrin) [inline src/services/abstractions/autofill.service.ts]: we don't want services having dependencies on UI components.

File: src/services/autofill.service.ts [hunk 3]
```diff
@@ -9,7 +9,7 @@ import AutofillScript from '../models/autofillScript';
 
 import { BrowserApi } from '../browser/browserApi';
 
-import { AutofillService as AutofillServiceInterface } from './abstractions/autofill.service';
+import { AutofillService as AutofillServiceInterface, IAutoFillOptions } from './abstractions/autofill.service';
```
Comment (kspearrin) [inline src/services/autofill.service.ts]: Should be multi-lined like enums import above

File: src/services/autofill.service.ts [hunk 4]
```diff
@@ -19,6 +19,9 @@ import {
 
 import { EventService } from 'jslib/abstractions/event.service';
 import { EventType } from 'jslib/enums/eventType';
+import { CardView, CipherView, IdentityView } from 'jslib/models/view';
+import AutofillForm from 'src/models/autofillForm';
```
Comment (kspearrin) [inline src/services/autofill.service.ts]: What is this?

**Comment by eliykat** [comment]
Hi @MarkKoester, thanks for your submission! I'm currently reviewing and cleaning up old pull requests. If you're still interested in progressing this PR, please:
* sign the Contributor License Agreement (see the link provided by the CLAassistant bot above)
* review/respond to any outstanding change requests or questions
* reply here and tag me (@) so that I can mark it for further review

As this PR is more than 12 months old, I'll close it for now. However, this is just housekeeping. We encourage our community members to get involved and make contributions, so you can comment here even after it's closed and I'll reopen it if required.

**Pull request closed without merging**